mautic/mautic
# Upsert functionality

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [No ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ No] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://github.com/acquia/mc-cs-plugin-custom-objects/issues/288



#### Description:


This PR adds the ability to 'upsert' an entity using 'INSERT INTO ... VALUES ... ON DUPLICATE KEY UPDATE' query. It also adds methods to check if the entity has been updated or inserted. 

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. See https://github.com/acquia/mc-cs-plugin-custom-objects/issues/288




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11654"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
